### PR TITLE
fix(#152): migrate toml package

### PR DESCRIPTION
**Pull request by wweir**
migrate toml package from [BurntSushi/toml](https://github.com/BurntSushi/toml) to [github.com/pelletier/go-toml/v2](https://github.com/pelletier/go-toml)

### CODE REVIEW

File: aconfigtoml/go.mod [hunk 1]
```diff
@@ -1,8 +1,10 @@
 module github.com/cristalhq/aconfig/aconfigtoml
 
-go 1.16
+go 1.21.0
```
Comment (cristaloleg) [inline aconfigtoml/go.mod:3]: Do we need this bump?

Comment (wweir) [reply aconfigtoml/go.mod:3]: It's generated automatically.
The [pelletier/go-toml](https://github.com/pelletier/go-toml/blob/af236b689f5ac1bdbdfeb8af1ed7b398a2dbe05e/go.mod#L3) declare the golang version

Comment (wweir) [reply aconfigtoml/go.mod:3]: If you know the underlying reason of [BurntSushi/toml](https://github.com/BurntSushi/toml), we won't need to exchange it.

**Comment by wweir** [comment]
There are still some other issues

**Pull request closed without merging**